### PR TITLE
feat(stats): Add Project Selector to Stats page

### PR DESCRIPTION
**Pull request by leeandher**
See [ER-1203](https://getsentry.atlassian.net/browse/ER-1203)

This PR refactors the organization statistics page to support project selection!

Demo: 

https://user-images.githubusercontent.com/35509934/196281200-55ac6658-0fa6-4155-bc9b-bca82a256b1e.mov

With and without flag:

https://user-images.githubusercontent.com/35509934/196281152-0c347827-35bf-42b2-b2ef-36932e6520e8.mov

TODO: 
- [x] Merge https://github.com/getsentry/sentry/pull/39955
- [x] Add video and screenshots
- [x] Implement a feature flag

### CODE REVIEW

**Comment by nhsiehgit** [comment]
A bit odd how we can filter the projects on the page via the global header, but then we can also filter again via the search box 🤔

File: static/app/views/organizationStats/usageStatsOrg.tsx
Comment (nhsiehgit) [inline static/app/views/organizationStats/usageStatsOrg.tsx:61]: do we need to utilize lodash here?

Ideally we should prefer not to utilize lodasdh unless absolutely necessary.

File: static/app/views/organizationStats/usageStatsProjects.tsx
Comment (nhsiehgit) [inline static/app/views/organizationStats/usageStatsProjects.tsx:114]: `projects: ...`?

File: static/app/views/organizationStats/index.tsx
Comment (nhsiehgit) [inline static/app/views/organizationStats/index.tsx:333]: 🤔 
personally i dislike this.
It's unexpected to have the page perform so differently while i'm still on teh same page/route/tab...

I would prefer to go back to the drawing board wtih this and figure out a better flow :(

**Review by nhsiehgit — COMMENTED** [review]
:( i'm not a huge fan tbh...

but i wont block on it. will leave it up to design i suppose

File: static/app/views/organizationStats/usageStatsOrg.tsx
Comment (leeandher) [reply static/app/views/organizationStats/usageStatsOrg.tsx:61]: Sure, I used lodash here since `prevProjectIds !== currProjectIds` can never return true since they're objects.

File: static/app/views/organizationStats/usageStatsProjects.tsx
Comment (leeandher) [reply static/app/views/organizationStats/usageStatsProjects.tsx:114]: The API uses `project` not `projects`

File: static/app/views/organizationStats/index.tsx
Comment (leeandher) [reply static/app/views/organizationStats/index.tsx:333]: I still prefer this to having pages show similar information but be accessible through different routes. E.g. If we are keeping org stats around, but showing that same graph somewhere else like `stats/project/<project-slug>/` and push the user to that new route.

Then we have almost the same UI served from two different areas in the app, and we have to worry about how to direct users to one versus the other, and navigating between them.
I personally don't see the harm since we can surface the differences elsewhere in the app by linking directly to `stats/?project=1` or adding placeholder boxes which say `To view [X] data, select a single project`.

cc @jmanhart Curious to hear your thoughts on this design as well!

**Review by leedongwei — APPROVED** [review]
Code is LGTM. Up to design/product for final decision if this should make it in.

File: static/app/views/organizationStats/index.spec.tsx
Comment (leeandher) [reply static/app/views/organizationStats/index.spec.tsx]: good catch! thanks

**Pull request merged**
